lms/djangoapps/save_for_later/api/v1/views.py

```diff
@@ -0,0 +1,116 @@
+"""
+Save for later views
+"""
+
+
+import logging
+
+from django.conf import settings
+from django.utils.decorators import method_decorator
+from django.contrib.sites.models import Site
+from ratelimit.decorators import ratelimit
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from django.db import transaction
+from opaque_keys.edx.keys import CourseKey
+from edx_ace import ace
+from edx_ace.recipient import Recipient
+
+from common.djangoapps.track import segment
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.core.djangoapps.user_api.accounts.api import get_email_validation_error
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.core.djangoapps.ace_common.template_context import get_base_template_context
+
+from lms.djangoapps.save_for_later.message_types import SaveForLater
+from lms.djangoapps.save_for_later.models import SavedCourse
+
+log = logging.getLogger(__name__)
+
+POST_EMAIL_KEY = 'post:email'
+REAL_IP_KEY = 'openedx.core.djangoapps.util.ratelimit.real_ip'
+
+
+class SaveForLaterApiView(APIView):
+    """
+    API VIEW
+    """
+    @transaction.atomic
+    @method_decorator(ratelimit(key=POST_EMAIL_KEY, rate=settings.SAVE_FOR_LATER_EMAIL_RATE_LIMIT, method='POST'))
+    @method_decorator(ratelimit(key=REAL_IP_KEY, rate=settings.SAVE_FOR_LATER_IP_RATE_LIMIT, method='POST'))
+    def post(self, request):
+        """
+        **Use Case**
+
+            * Send favorite course through email to user for later learning.
+
+        **Example Request**
+
+            POST /api/v1/save/course/
+
+        **Example POST Request**
+
+            {
+                "email": "test@edx.org",
+                "course_id": "course-v1:edX+DemoX+2021",
+                "marketing-url": "https://test.com",
+                "org_img_url": "https://test.com/logo.png"
+
+            }
+        """
+        user = request.user
+        course_id = request.POST.get('course_id')
+        email = request.POST.get('email')
+        marketing_url = request.POST.get('marketing_url')
+        org_img_url = request.POST.get('org_img_url')
+
+        course_key = CourseKey.from_string(course_id)
+
+        if getattr(request, 'limited', False):
+            return Response({'result': 'failure'}, status=403)
```

Comment:
can we send error code in our API response like we do for our Authn APIs
```suggestion
            return Response({'error_code': 'rate-limited`}, status=403)
```

Reply:
@zainab-amir @attiyaIshaque is working on it. She will update response according to need.